publisher-sdk/src/test/java/com/criteo/publisher/advancednative/AdChoiceOverlayNativeRendererTest.kt

```diff
@@ -53,6 +58,8 @@ class AdChoiceOverlayNativeRendererTest {
   @Before
   fun setUp() {
     renderer = AdChoiceOverlayNativeRenderer(delegate, adChoiceOverlay)
+    // Main Lopper is null for unit tests, so just mock executor to avoid NullPointerException
+    whenever(mockedDependenciesRule.dependencyProvider.provideRunOnUiThreadExecutor()).thenReturn(mock())
```

Comment:
This is true for all Java tests. Instead of having a particular mock on this test, what about setting this directly in the MockedDependenciesRule so that a java test only needs to setup it to have all env mocked correctly.

For instance, for Application, the rule is setting up a mock Application via this class: https://github.com/criteo/android-publisher-sdk/blob/main/test-utils/src/main/java/com/criteo/publisher/application/ApplicationResource.java
Only for java tests: https://github.com/criteo/android-publisher-sdk/blob/main/test-utils/src/main/java/com/criteo/publisher/application/InstrumentationUtil.java#L33

For the UI thread executor, maybe could you mock it the same way by extending/reusing this component: https://github.com/criteo/android-publisher-sdk/blob/main/test-utils/src/main/java/com/criteo/publisher/concurrent/MultiThreadResource.java